File: Programmers/codingTest_Kit/Sort/42746.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool comp(string a, string b){
    return a + b > b + a;
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> tmp;
    for(auto i : numbers){
        tmp.push_back(to_string(i));
    }
    sort(tmp.begin(), tmp.end(), comp);
    if(tmp[0] == "0") return "0";
    for(auto i : tmp) answer += i;
    return answer;
}

File: Programmers/codingTest_Kit/Sort/42746.java
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";

        List<Integer> list = Arrays.stream(numbers).mapToObj(i -> (Integer) i).collect(Collectors.toList());

        Collections.sort(list, (a, b) -> {
            String as = String.valueOf(a), bs = String.valueOf(b);
            return -Integer.compare(Integer.parseInt(as + bs), Integer.parseInt(bs + as));
        });

        StringBuilder sb = new StringBuilder();
        for (Integer i : list) {
            sb.append(i);
        }
        answer = sb.toString();
        if (answer.charAt(0) == '0') {
            return "0";
        } else {
            return answer;
        }
    }
}